docs/querying/api.md

```diff
@@ -1031,6 +1031,36 @@ $ curl http://localhost:9090/api/v1/status/tsdb
 
 *New in v2.15*
 
+### WAL Replay Stats
+
+The following endpoint returns information about the WAL replay:
+
+```
+GET /api/v1/status/walreplay
+```
+
+**first**: The first segment needed to be replayed.
+**last**: The last segment needed to be replayed.
+**read**: The latest segment replayed.
```

Comment:
`read` is the most recent segment to be replayed. So if we start at the 5th segment, then read it, the `read` value would be 5. In the front-end code, the `first` value is subtracted from the `read` value to generate the progress bar. I now realize this is flawed because the `read` value can be the same as the `first` value, showing no progress, even though that just means that only the first segment _has_ been read. The `read` value not being set to the starting value is what caused the bugs that @roidelapluie found. I think I'll probably need to rework this, maybe just making it the total number of segments read regardless of the ID, starting from 0, and then let the client (or server) figure the progress out.

Reply:
Furthermore, the reason that all of these fields are so specific (as to a single number counting from 0) is that I wanted this endpoint to provide the same data as the log lines do.

Reply:
Yea, I think we don't need to be strict towards log line, will look how it looks now